File: online judge.kaner murillo/python/514.py
#https://onlinejudge.org/external/5/514.pdf
#514

from sys import stdin
from collections import deque

cantidad = int(stdin.readline().strip())

while cantidad!= 0:

    vagones = [int(x) for x in stdin.readline().split()]

    while vagones != [0]:

        i = 0
        n = 1
        estacion = deque()
        ans = True

        while(n<len(vagones)+1):

            estacion.append(n)

            if(estacion[-1] == vagones[i]):

                while len(estacion) != 0 and estacion[-1] == vagones[i]:

                    estacion.pop()
                    i+=1
            
            n+=1


        while len(estacion)!= 0 and ans == True:

            if(estacion[-1] == vagones[i]):

                estacion.pop()
                i+=1

            else:
                ans = False

        
        if ans == True:

            print("Yes")
        
        else:
            print("No")

        vagones = [int(x) for x in stdin.readline().split()]

    print()

    cantidad = int(stdin.readline().strip())

File: online judge.kaner murillo/c++/514.cpp
//https://onlinejudge.org/external/5/514.pdf
//514

#include<stack>
#include<cstdio>
#include<iostream>
#include<string>
#include<sstream>
#include<vector>

using namespace std;

int main(){

    string cadena;
    int v,cantidad,i,n;
    vector<int>orden;

    bool ans;

    while(scanf("%i", &cantidad) && cantidad!=0){

        cin.ignore();

        while(getline(cin,cadena) && cadena != "0"){

            orden.clear();

            stack<int>estacion;
        
            stringstream stk(cadena);
            ans = true;

            while(stk>>v){
                orden.push_back(v);
            }

            i = 0;
            n = 1;

            
            while(n<orden.size()+1){

                estacion.push(n);

                if(estacion.top() == orden[i]){

                    while(!estacion.empty() && estacion.top() == orden[i]){

                        estacion.pop();
                        i+=1;

                    }    
                }

                n+=1;
            }

            while(!estacion.empty() && ans == true){
                
                if(estacion.top() == orden[i]){

                    estacion.pop();
                    i+=1;
                }

                else{
                    ans = false;
                }
                
            }

            if(ans == true){

                printf("Yes\n");
            }

            else{
                printf("No\n");
            }
        }

        printf("\n");
    }

    return 0;
}